src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java

```diff
@@ -378,6 +401,253 @@ public QuizExerciseResource(QuizExerciseRepository quizExerciseRepository, Parti
+        for (Question question : quizExercise.getQuestions()) {
+            //check if the question were already in the originalQuizExercise -> if not it's an added question
+            if (originalQuizExercise.getQuestions().contains(question)) {
+                // find original unchanged question
+                Question originalQuestion = originalQuizExercise.findQuestionById(question.getId());
+                //reset score (not allowed to change)
+                question.setScore(originalQuestion.getScore());
+                //reset invalid if the question is already invalid;
+                question.setInvalid(question.isInvalid() || originalQuestion.isInvalid());
+
+                // check if a question is  set invalid or if the scoringType has changed
+                // if true an update of the Statistics and Results is necessary
+                updateOfResultsAndStatisticsNecessary = updateOfResultsAndStatisticsNecessary ||
+                    (question.isInvalid() && !originalQuestion.isInvalid()) ||
+                    !question.getScoringType().equals(originalQuestion.getScoringType());
+
+                //undo all not allowed changes in the answers of the MultipleChoiceQuestion
+                // and check if the answers effect make an update of the statistics and results necessary
+                if (question instanceof MultipleChoiceQuestion) {
+                    updateOfResultsAndStatisticsNecessary = updateOfResultsAndStatisticsNecessary ||
+                        undoUnallowedChangesAndCheckMultipleChoiceQuestionAnswers((MultipleChoiceQuestion) question, (MultipleChoiceQuestion) originalQuestion);
+                }
+
+                if (question instanceof DragAndDropQuestion) {
+                    // TODO: @Moritz: check changes in DragAndDropQuestions
+                }
+
+            } else {
+                // question is added (not allowed), mark question for remove
+                addedQuestions.add(question);
+            }
+        }
+        // remove all added questions
+        quizExercise.getQuestions().removeAll(addedQuestions);
+
+        // check if an question was deleted (not allowed added quistions are not relevant)
+        // if true an update of the Statistics and Results is necessary
+        if (quizExercise.getQuestions().size() != originalQuizExercise.getQuestions().size()) {
+            updateOfResultsAndStatisticsNecessary = true;
+        }
+        return updateOfResultsAndStatisticsNecessary;
+    }
+
+    /**
+     * 1. undo all changes which are not allowed ( adding Answers)
+     * 2. check if an update of the Results and Statistics is necessary
+     *
+     * @param question the changed MultipleChoiceQuestion-object
+     * @param originalQuestion the original MultipleChoiceQuestion-object, which will be compared with the new changed question
+     *
+     * @return a boolean which is true if the answer-changes make an update necessary and false if not
+     */
```

Comment:
the name of this method is mislead: does it have 1 purpose or 2?
Please move the method into a service or into a domain class.